Error in building MATLAB wrapper: No module named 'gtwrap'

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

## Description

<!-- A clear description of the bug -->
I am trying to build the MATLAB wrapper. Getting the following error
```
Traceback (most recent call last):
  File "/home/sarvesh/Libraries/gtsam/wrap/scripts/matlab_wrap.py", line 11, in <module>
    from gtwrap.matlab_wrapper import MatlabWrapper
ModuleNotFoundError: No module named 'gtwrap'
matlab/CMakeFiles/gtsam_matlab_wrapper.dir/build.make:86: recipe for target 'wrap/gtsam/gtsam_wrapper.cpp' failed
make[2]: *** [wrap/gtsam/gtsam_wrapper.cpp] Error 1
CMakeFiles/Makefile2:30583: recipe for target 'matlab/CMakeFiles/gtsam_matlab_wrapper.dir/all' failed
make[1]: *** [matlab/CMakeFiles/gtsam_matlab_wrapper.dir/all] Error 2
Makefile:165: recipe for target 'all' failed
make: *** [all] Error 2
```

## Steps to reproduce

1. 
```
cmake -DCMAKE_BUILD_TYPE=Release -DGTSAM_TOOLBOX_INSTALL_PATH:PATH=$HOME/matlab/toolbox  -DGTSAM_BUILD_UNSTABLE:OPTION=ON  -DGTSAM_INSTALL_MATLAB_TOOLBOX=ON ..
```
2. `make`

<!-- If you have a code sample, unit test, error messages, stack traces, etc., please provide it here as well -->


## Environment
Ubuntu 18.04
gtsam 4.1.1
MATLAB 2018a
Python 3.8.12 (conda env)

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

<!-- The more the information, the faster we can help resolve the issue -->

## Additional information
I am able to build the python wrapper just fine
```
cmake -DCMAKE_BUILD_TYPE=Release -DGTSAM_BUILD_UNSTABLE:OPTION=ON -DGTSAM_PYTHON_VERSION=3.8.12 -DGTSAM_BUILD_PYTHON=1 ..
```
<!-- Add any other infor or context about the problem here. -->

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -13,7 +13,7 @@ $ make install
 ## Important Installation Notes
 
 1. GTSAM requires the following libraries to be installed on your system:
-    - BOOST version 1.65 or greater (install through Linux repositories or MacPorts). Please see [Boost Notes](#boost-notes).
+    - BOOST version 1.65 or greater (install through Linux repositories or MacPorts). Please see [Boost Notes](#boost-notes) for version recommendations based on your compiler.
 
     - Cmake version 3.0 or higher
     - Support for XCode 4.3 command line tools on Mac requires CMake 2.8.8 or higher
@@ -72,7 +72,7 @@ execute commands as follows for an out-of-source build:
 Versions of Boost prior to 1.65 have a known bug that prevents proper "deep" serialization of objects, which means that objects encapsulated inside other objects don't get serialized.
 This is particularly seen when using `clang` as the C++ compiler.
 
-For this reason we require Boost>=1.65, and recommend installing it through alternative channels when it is not available through your operating system's primary package manager.
+For this reason we recommend Boost>=1.65, and recommend installing it through alternative channels when it is not available through your operating system's primary package manager.
 
 ## Known Issues
 

--- a/gtsam/nonlinear/LevenbergMarquardtParams.h
+++ b/gtsam/nonlinear/LevenbergMarquardtParams.h
@@ -35,7 +35,7 @@ class LevenbergMarquardtOptimizer;
 class GTSAM_EXPORT LevenbergMarquardtParams: public NonlinearOptimizerParams {
 
 public:
-  /** See LevenbergMarquardtParams::lmVerbosity */
+  /** See LevenbergMarquardtParams::verbosityLM */
   enum VerbosityLM {
     SILENT = 0, SUMMARY, TERMINATION, LAMBDA, TRYLAMBDA, TRYCONFIG, DAMPED, TRYDELTA
   };

--- a/matlab/CMakeLists.txt
+++ b/matlab/CMakeLists.txt
@@ -92,8 +92,8 @@ if(GTSAM_UNSTABLE_INSTALL_MATLAB_TOOLBOX)
   endif()
 
   # Wrap
-  matlab_wrap(${GTSAM_SOURCE_DIR}/gtsam_unstable/gtsam_unstable.i "gtsam" ""
-              "${mexFlags}" "${ignore}")
+  matlab_wrap(${GTSAM_SOURCE_DIR}/gtsam_unstable/gtsam_unstable.i "gtsam_unstable"
+              "${GTSAM_ADDITIONAL_LIBRARIES}" "" "${mexFlags}" "${ignore}")
 endif(GTSAM_UNSTABLE_INSTALL_MATLAB_TOOLBOX)
 
 # Record the root dir for gtsam - needed during external builds, e.g., ROS

--- a/wrap/gtwrap/pybind_wrapper.py
+++ b/wrap/gtwrap/pybind_wrapper.py
@@ -412,7 +412,7 @@ class PybindWrapper:
 
     def wrap_instantiated_declaration(
             self, instantiated_decl: instantiator.InstantiatedDeclaration):
-        """Wrap the class."""
+        """Wrap the forward declaration."""
         module_var = self._gen_module_var(instantiated_decl.namespaces())
         cpp_class = instantiated_decl.to_cpp()
         if cpp_class in self.ignore_classes:
@@ -420,7 +420,7 @@ class PybindWrapper:
 
         res = (
             '\n    py::class_<{cpp_class}, '
-            '{shared_ptr_type}::shared_ptr<{cpp_class}>>({module_var}, "{class_name}")'
+            '{shared_ptr_type}::shared_ptr<{cpp_class}>>({module_var}, "{class_name}");'
         ).format(shared_ptr_type=('boost' if self.use_boost else 'std'),
                  cpp_class=cpp_class,
                  class_name=instantiated_decl.name,

--- a/wrap/matlab.h
+++ b/wrap/matlab.h
@@ -37,15 +37,16 @@ extern "C" {
 #include <mex.h>
 }
 
-#include <boost/shared_ptr.hpp>
+#include <boost/cstdint.hpp>
 #include <boost/make_shared.hpp>
+#include <boost/shared_ptr.hpp>
 
 #include <list>
-#include <string>
-#include <sstream>
-#include <typeinfo>
 #include <set>
+#include <sstream>
 #include <streambuf>
+#include <string>
+#include <typeinfo>
 
 using namespace std;
 using namespace boost; // not usual, but for conciseness of generated code

--- a/wrap/tests/expected/python/class_pybind.cpp
+++ b/wrap/tests/expected/python/class_pybind.cpp
@@ -105,7 +105,7 @@ PYBIND11_MODULE(class_py, m_) {
                         return redirect.str();
                     }, py::arg("s") = "factor: ", py::arg("keyFormatter") = gtsam::DefaultKeyFormatter);
 
-    py::class_<SuperCoolFactor<gtsam::Pose3>, std::shared_ptr<SuperCoolFactor<gtsam::Pose3>>>(m_, "SuperCoolFactorPose3")
+    py::class_<SuperCoolFactor<gtsam::Pose3>, std::shared_ptr<SuperCoolFactor<gtsam::Pose3>>>(m_, "SuperCoolFactorPose3");
 
 #include "python/specializations.h"
 

--- a/wrap/tests/fixtures/inheritance.i
+++ b/wrap/tests/fixtures/inheritance.i
@@ -1,6 +1,5 @@
 // A base class
 virtual class MyBase {
-
 };
 
 // A templated class

